Breaking change docs: CborReader/CborWriter enforce default maximum nesting depth (.NET 11) (#54360)
dotnet/docs: docs/core/compatibility/core-libraries/11/cbor-max-depth.md

Reviewer: @copilot Why is this under extensions and not core libraries? Also, please use xrefs for the first mentions of CborReader and CborWriter.

## Recommended action

If your application processes CBOR data nested more deeply than the defaults allow (64 levels for reading, 1000 levels for writing), use the new options types to specify a larger limit:

```csharp
// For reading CBOR data nested more than 64 levels deep
var options = new CborReaderOptions { MaxDepth = 256 };
var reader = new CborReader(data, options);
```

```csharp
// For writing CBOR data nested more than 1000 levels deep
var writerOptions = new CborWriterOptions { MaxDepth = 2000 };
var writer = new CborWriter(writerOptions);
```

> [!NOTE]
> Unlike `Utf8JsonReader`/`Utf8JsonWriter`, setting `MaxDepth = 0` in CBOR means *no nesting is allowed* (not "use the runtime default"). To use the runtime default, set `MaxDepth = -1` or omit the property when constructing `CborReaderOptions`/`CborWriterOptions`.

> You cannot use an AppContext switch to restore the previous unlimited-depth behavior.

Reviewer: ```suggestion
There is no AppContext switch to restore the previous unlimited-depth behavior.
```

```

> [!NOTE]
> Unlike `Utf8JsonReader`/`Utf8JsonWriter`, setting `MaxDepth = 0` in CBOR means *no nesting is allowed* (not "use the runtime default"). To use the runtime default, set `MaxDepth = -1` or omit the property when constructing `CborReaderOptions`/`CborWriterOptions`.

There is no AppContext switch to restore the previous unlimited-depth behavior.

## Affected APIs

- <xref:System.Formats.Cbor.CborReader.ReadStartArray?displayProperty=fullName>
- <xref:System.Formats.Cbor.CborReader.ReadStartMap?displayProperty=fullName>
- <xref:System.Formats.Cbor.CborReader.ReadStartIndefiniteLengthByteString?displayProperty=fullName>
- <xref:System.Formats.Cbor.CborReader.ReadStartIndefiniteLengthTextString?displayProperty=fullName>
- <xref:System.Formats.Cbor.CborWriter.WriteStartArray(System.Nullable{System.Int32})?displayProperty=nameWithType>
- <xref:System.Formats.Cbor.CborWriter.WriteStartIndefiniteLengthByteString?displayProperty=fullName>
- <xref:System.Formats.Cbor.CborWriter.WriteStartIndefiniteLengthTextString?displayProperty=fullName>
- <xref:System.Formats.Cbor.CborWriter.WriteStartMap(System.Nullable{System.Int32})?displayProperty=fullName>

> 

Reviewer: ```suggestion
```
